### PR TITLE
added TTC4900: A Benchmark Data for Turkish Text Categorization dataset

### PR DESCRIPTION
**Pull request by yavuzKomecoglu**
This PR adds the TTC4900 dataset which is a Turkish Text Categorization dataset by me and @basakbuluz. 

Homepage: [https://www.kaggle.com/savasy/ttc4900](https://www.kaggle.com/savasy/ttc4900)
Point of Contact: [Savaş Yıldırım](mailto:savasy@gmail.com) / @savasy

### CODE REVIEW

**Comment by yavuzKomecoglu** [comment]
@lhoestq, can you help with creating dummy_data?

**Review by lhoestq — COMMENTED** [review]
We need to have a file named `dummy_data.zip` in the folder `./datasets/ttc4900/dummy/ttc4900/1.0.0`

To build this zip file, follow those steps:
- first go inside `./datasets/ttc4900/dummy/ttc4900/1.0.0` and create a folder named `dummy_data`
- in the `dummy_data` folder create a csv file named `7allV03.csv` and add a few examples from the original csv file in it (for example copy the first 5 lines)
- then zip the `dummy_data` folder into dummy_data.zip

To make sure it works as expected please run the dummy data test:
```
RUN_SLOW=1 pytest tests/test_dataset_common.py::LocalDatasetTest::test_load_dataset_all_configs_ttc4900
```

Let me know if that worked :)

File: datasets/ttc4900/ttc4900.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+
+_DESCRIPTION = """\
+The data set is taken from kemik group
+http://www.kemik.yildiz.edu.tr/
+The data are pre-processed for the text categorization, collocations are found, character set is corrected, and so forth.
+We named TTC4900 by mimicking the name convention of TTC 3600 dataset shared by the study http://journals.sagepub.com/doi/abs/10.1177/0165551515620551
+"""
+
+_CITATION = ""
```
Comment (lhoestq) [inline datasets/ttc4900/ttc4900.py:34]: is there a citation we can add here ?

**Review by lhoestq — COMMENTED** [review]
The code looks all good to me btw, thanks !

Feel free to ping me if you have questions about the dummy data or when everything is ready for the final review

**Comment by lhoestq** [comment]
Hi @yavuzKomecoglu did you manage to build the dummy data ?

**Comment by yavuzKomecoglu** [comment]
> Hi @yavuzKomecoglu did you manage to build the dummy data ?

Hi, sorry for the return. I've created dummy_data.zip manually.

File: datasets/ttc4900/ttc4900.py (hunk 1 above)
Comment (yavuzKomecoglu) [reply datasets/ttc4900/ttc4900.py:34]: I will contact the original owner of the dataset and update it later.

File: datasets/ttc4900/README.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+## Dataset Structure
+
+### Data Instances
+
+A text classification dataset with 7 different news category. 
+
+### Data Fields
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/ttc4900/README.md:71]: Here can you add a brief description of each data field please `text` and `category` please ?

File: datasets/ttc4900/README.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+### Languages
+
+The dataset is based on Turkish.
+
+## Dataset Structure
+
+### Data Instances
+
+A text classification dataset with 7 different news category.
```
Comment (lhoestq) [inline datasets/ttc4900/README.md:67]: here can you add one example from the dataset ?

You can get one by using `load_dataset` and then take the first sample with `dataset["train"][0]`

File: datasets/ttc4900/ttc4900.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN, gen_kwargs={"filepath": os.path.join(path_to_manual_file, _FILENAME)}
+            )
+        ]
+
+    def _generate_examples(self, filepath):
+        """Generate TTC4900 examples."""
+        logging.info("⏳ Generating examples from = %s", filepath)
+        print("filepath", filepath)
```
Comment (lhoestq) [inline datasets/ttc4900/ttc4900.py:112]:
```suggestion
```
can you remove this print please ?

**Review by lhoestq — COMMENTED** [review]
Nice thank you !

Before we merge can you fill the two sections of the dataset card I suggested ?
And also remove one remaining print statement

File: datasets/ttc4900/README.md (hunk 3 above)
Comment (yavuzKomecoglu) [reply datasets/ttc4900/README.md:67]: > here can you add one example from the dataset ?
> 
> You can get one by using `load_dataset` and then take the first sample with `dataset["train"][0]`

I updated your suggestions. Thank you very much for your support.

**Comment by yavuzKomecoglu** [comment]
> Nice thank you !
> 
> Before we merge can you fill the two sections of the dataset card I suggested ?
> And also remove one remaining print statement

I updated your suggestions. Thank you very much for your support.

**Comment by lhoestq** [comment]
I think you accidentally pushed the readme of another dataset (name_to_nation).
I removed it so you have to `git pull`

Because of that I guess your changes about the ttc4900 was not included.
Feel free to ping me once they're added

**Comment by yavuzKomecoglu** [comment]
> I think you accidentally pushed the readme of another dataset (name_to_nation).
> I removed it so you have to `git pull`
> 
> Because of that I guess your changes about the ttc4900 was not included.
> Feel free to ping me once they're added



> I think you accidentally pushed the readme of another dataset (name_to_nation).
> I removed it so you have to `git pull`
> 
> Because of that I guess your changes about the ttc4900 was not included.
> Feel free to ping me once they're added

I did `git pull` and updated readme ttc4900.

File: datasets/ttc4900/README.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+## Dataset Structure
+
+### Data Instances
+
+A text classification dataset with 7 different news category. 
+
+**Category:** siyaset
+
+**Text:** paris teki infaz imralı ile başlayan sürece bir darbe mi elif_çakır ın sunduğu söz_bitmeden in bugünkü konuğu gazeteci melih altınok oldu programdan satıbaşları imralı ile görüşmeler hangi aşamada bundan sonra ne olacak hangi kesimler sürece engel oluyor psikolojik mayınlar neler türk solu bu dönemde evrensel sorumluluğunu yerine getirebiliyor mu elif_çakır sordu melih altınok söz_bitmeden de yanıtladı elif_çakır pkk nın silahsızlandırılmasına yönelik olarak öcalan ile görüşme sonrası 3 kadının infazı enteresan çünkü kurucu isimlerden birisi sen nasıl okudun bu infazı melih altınok herkesin ciddi anlamda şüpheleri var şu an yürüttüğümüz herşey bir delile dayanmadığı için komple teorisinden ibaret kalacak ama şöyle bir durum var imralı görüşmelerin ilk defa bir siyasi iktidar tarafından açıkça söylendiği bir dönem ardından geliyor bu sürecin gerçekleşmemesini isteyen kesimler yaptırmıştır dedi
```
Comment (lhoestq) [inline datasets/ttc4900/README.md:71]:
````suggestion
Here is an example from the dataset:

```
{
"category": 0,  # politics/siyaset
"text": "paris teki infaz imralı ile başlayan sürece bir darbe mi elif_çakır ın sunduğu söz_bitmeden in bugünkü konuğu gazeteci melih altınok oldu programdan satıbaşları imralı ile görüşmeler hangi aşamada bundan sonra ne olacak hangi kesimler sürece engel oluyor psikolojik mayınlar neler türk solu bu dönemde evrensel sorumluluğunu yerine getirebiliyor mu elif_çakır sordu melih altınok söz_bitmeden de yanıtladı elif_çakır pkk nın silahsızlandırılmasına yönelik olarak öcalan ile görüşme sonrası 3 kadının infazı enteresan çünkü kurucu isimlerden birisi sen nasıl okudun bu infazı melih altınok herkesin ciddi anlamda şüpheleri var şu an yürüttüğümüz herşey bir delile dayanmadığı için komple teorisinden ibaret kalacak ama şöyle bir durum var imralı görüşmelerin ilk defa bir siyasi iktidar tarafından açıkça söylendiği bir dönem ardından geliyor bu sürecin gerçekleşmemesini isteyen kesimler yaptırmıştır dedi"
}
```
````

**Review by lhoestq — APPROVED** [review]
Thank you !

**Comment by lhoestq** [comment]
merging since the Ci is fixed on master

**Pull request merged**